wordpress-mobile/WordPress-iOS
# AztecPostViewController: Preventing post removal

### Details:
The Hybrid Editor proceeds to "removing" the post being edited (whenever `discardChanges` runs), only if we "Own" the post:

```
    if (self.ownsPost) {
        [self.post remove];
        self.post = nil;
    }
```

Note that the flag `_ownsPost` is set to **YES** whenever the Editor is initialized with a post that's already a draft. In our Aztec's port, we've missed the `ownsPost` check, and thus, the Post was being removed, always.

IMHO if the Post is to be deleted, it's the called (responsible for creating the Draft passed on to the Editor), the one who should be responsible for doing so. In which case, i'm hereby nuking the entire `post.remove()` call.

Fixes #6534

### Testing:
1. Make sure Aztec is Enabled
2. Open **My Sites** > **(Any) Site**
3. Open the Post List and filter by **Drafts**
4. Edit any Draft
5. Dismiss the editor, without making any changes.

Verify that the post you've pressed in step (3) is still visible.

Needs review: @astralbodies 
cc @diegoreymendez
